### PR TITLE
Locked ptr test, documentation, and example

### PR DESCRIPTION
**Pull request by henrygerardmoore**
Because of #23, I realized that we don't actually test locked ptr, so I made this PR. I also added a non-blocking option to the constructor so you can see if you can get the locked ptr.

### CODE REVIEW

**Comment by facontidavide** [comment]
I Will have a look, thanks

**Comment by henrygerardmoore** [comment]
> I Will have a look, thanks

Thanks! Feel free to push directly to this branch

File: data_tamer_cpp/include/data_tamer/logged_value.hpp [hunk 1]
```diff
@@ -49,15 +49,22 @@ class LoggedValue
...
   ///   *ref += 1;
   /// }
   [[nodiscard]] LockedPtr<T> getLockedPtr();
 
+  /**
+   * @brief This method is the same as getLockedPtr but does not block
+   *
+   * @return LockedPtr<T>
+   */
+  [[nodiscard]] LockedPtr<T> tryGetLockedPtr();
```
Comment (facontidavide) [inline data_tamer_cpp/include/data_tamer/logged_value.hpp:66]: I have a problem withg this.

Conceptually, `tryLock` should mean "sorry, I was not able to lock it, and I am telling you that it is NOT locked".

Instead here it is more like "you will always get a result, doesn't matter if it is unsafe or not".

What you implemented here is `getUnsafePtr()`

**Comment by facontidavide** [comment]
Sorry, but I don't believe this should be merged.

The right thing to do is to upgrade the locked reference class to this: https://github.com/facontidavide/ProtectedValue

What we really want to are Mutable and Constant reference, that are wrappers to read/write mutexes

**Review by facontidavide — CHANGES_REQUESTED** [review]

File: data_tamer_cpp/include/data_tamer/logged_value.hpp (hunk 1 above)
Comment (henrygerardmoore) [reply data_tamer_cpp/include/data_tamer/logged_value.hpp:66]: Should this return a `tl::expected` or something, instead? I'm also fine to just close this PR and upgrade to your `ProtectedValue` that you linked, if you think that would be better. We could also just remove the `tryGetLockedPtr` and make this PR just add a test and some documentation.

Comment (henrygerardmoore) [reply data_tamer_cpp/include/data_tamer/logged_value.hpp:66]: I have removed this entirely, as well as the changes to the LockedPtr constructor.

**Comment by henrygerardmoore** [comment]
@facontidavide I removed the unsafe pointer function, now this PR is just a test and example

**Pull request merged**